### DISCUSSION TITLE
How can I use `bit test` in my lint-staged Git pre-commit hook?

### DISCUSSION THREAD

**Discussion by dreamawakening** [teambit/bit#4629] [Help]
I have a `lint-staged` config for my Git pre-commit hook:

```js
module.exports = {
  "*.{ts,tsx,js}": [
    "prettier --write",
    "eslint -c .eslintrc.js --fix",
    "jest --bail --findRelatedTests",
  ],
  "*.md": ["remark -u preset-lint-recommended"],
};
```

This works fine in my other projects, because `jest` is the command to run tests. `lint-staged` passes a list of each changed file to `jest --bail --findRelatedTests`, so it only tests the files being committed (and not the entire project).

However, `bit test` is used in this environment. And this command doesn't have an argument like `--findRelatedTests` where it will only test the files being committed.

How should I handle this?

**Comment by itaymendel** [comment]
Hi,

The "related" flow is a core feature in Bit. By default all your operations (compile, build, test, lint, format...) operate only on modified components and their dependents (recursively). Potentially you can just run `bit test` and you get the same dev-experience without needing the additional configuration.

**Reply by dreamawakening** [reply]
@itaymendel the problem is `lint-staged` passes a list of all the changed files to the command (in this case `bit test`). 

So even though `bit test` only runs for modified components, this command breaks because `lint-staged` passes it a list of all of the modified files as an argument.

Therefore if an option can be added to `bit test` to pass the file path of each modified file, it will just work.
